packages/api/lambdas/sf-event-sqs-to-db-records.js

```diff
@@ -82,50 +119,124 @@ const shouldWriteExecutionToRDS = async (cumulusMessage, knex) => {
   }
 };
 
-const saveExecution = async (cumulusMessage, knex) => {
-  const executionModel = new Execution();
-  const executionArn = getMessageExecutionArn(cumulusMessage);
-
-  const isRDSWriteEnabled = await shouldWriteExecutionToRDS(cumulusMessage, knex);
+const saveExecutionViaTransaction = async ({ cumulusMessage, trx }) =>
+  trx(tableNames.executions)
+    .insert({
+      arn: getMessageExecutionArn(cumulusMessage),
+      cumulus_version: getMessageCumulusVersion(cumulusMessage),
+    });
 
-  if (!isRDSWriteEnabled) {
+const saveExecution = async ({
+  cumulusMessage,
+  knex,
+  executionModel = new Execution(),
+}) =>
+  knex.transaction(async (trx) => {
+    await saveExecutionViaTransaction({ cumulusMessage, trx });
     return executionModel.storeExecutionFromCumulusMessage(cumulusMessage);
-  }
+  });
 
-  try {
-    return await knex.transaction(async (trx) => {
-      await trx(tableNames.executions)
-        .insert({
-          arn: executionArn,
-          cumulus_version: getMessageCumulusVersion(cumulusMessage),
-        });
-      return executionModel.storeExecutionFromCumulusMessage(cumulusMessage);
+const savePdrViaTransaction = async ({
+  cumulusMessage,
+  collection,
+  provider,
+  trx,
+}) =>
+  trx(tableNames.pdrs)
+    .insert({
+      name: getMessagePdrName(cumulusMessage),
+      status: getWorkflowStatus(cumulusMessage),
+      collectionCumulusId: collection.cumulusId,
+      providerCumulusId: provider.cumulusId,
     });
-  } catch (error) {
-    log.error(`Failed to write execution records for ${executionArn}`, error);
-    throw error;
+
+const savePdr = async ({
+  cumulusMessage,
+  collection,
+  provider,
+  knex,
+  pdrModel = new Pdr(),
+}) => {
+  // If there is no PDR in the message, then there's nothing to do here, which is fine
+  if (!messageHasPdr(cumulusMessage)) {
+    return true;
```

Comment:
I don't like the weird/arbitrary return value here, but the idea was just that the function should return normally in this case

Reply:
Hmm.   Right now this return looks to be:  

Promise<PromiseResult<AWS.DynamoDB.DocumentClient.UpdateItemOutput, AWS.AWSError>>  || undefined || true 

But our call is an await without any sort of eval - is there a reason not to invert the conditional and just 
```
if(messageHasPdr) {  
  < do the thing >
   return something;
}
```
with/without an explicit return?  

just adding `return` would also resolve the promise as undefined.   I'm not sure what making the return truthy gets us here, unless you're planning on evaluating the 'undefined' response for reasons.

Reply:
Just using `return` causes an `inconsistent-return` error from ESlint. As does inverting the logic. I'm tempted to ignore ESlint as being capricious, except that I feel like it often surfaces poorly written code. So I'm not sure what to do here

Reply:
There's probably a legit case for `treatUndefinedAsUnspecified` being true, but I'm curious how the downstream code that's returning undefined is passing lint then.    🤔    Anyway, `null` is explicit, is that the right pattern for this sort of return, given that lint rule? 🤔

Reply:
I realized that we're going to want to explicitly return the PDR's cumulusId so that a granule can refer to it (implementation of that write logic in the next PDR), so this PDR write logic needed a bit of a refactor. Given the refactor, I think returning `undefined` when there is no PDR on the message is appropriate, so that the function either returns `undefined`, returns the PDR cumulusId, or throws an error. Also, using `undefined` over `null` because we have ESlint warnings on using `null`

29fb20d142f112e268888b6005cc4be50e484a4e